web/utils.py

```
@@ -414,6 +416,306 @@ def create_leaderboard_context(
+                    {
+                        "price": price_id,
+                    },
+                ],
+                payment_behavior="allow_incomplete",
+                metadata={"user_id": user.id, "plan_id": plan.id, "billing_period": billing_period},
+            )
+
+        # Update user membership
+        update_membership_from_subscription(user, subscription)
+
+        return {"success": True, "subscription": subscription}
```

Ruff (0.8.2)

Consider moving this statement to an `else` block

(TRY300)